### PR TITLE
Uptake listing detail api DAH-992

### PR DESCRIPTION
**Pull request by emilyjablonski**
[DAH-992](https://sfgovdt.jira.com/jira/software/c/projects/DAH/boards/122?modal=detail&selectedIssue=DAH-992)

The listing detail page is now using the api to fetch the individual listing, and some of the listing data (where possible due to remaining tickets) has been plugged into the skeleton.

The actual listing data you should see on the skeleton now:
* Listing card image
* Listing card name
* Listing card reserved tags
* Address under listing card
* Developer under listing card
* Orange reserved card above table
* Listing preference cards with titles
* Additional eligibility rules present if fields exist
* Building selection criteria link
* All listing features
* Additional fees deposit range and costs not included string
* Additional information cards
* Right panel application deadline status
* Right panel waitlist component with right waitlist numbers
* Right panel information session events
* Right panel open house events
* Right panel open listing lottery events

There are lots of pieces left / this is very much an iterative state. There is a lot we cannot show at all yet due to Bloom changes or remaining DAHLIA work. And even for the data pieces we could add above, some of them are not all going to look 100% visually as they should due to other Bloom changes. Additionally we are still not using translations / the right strings in some places, and that would be future cleanup to not inflate this work.

The biggest areas that are not able to show the right data still are: all of the tables (pricing, ami, occupancy) because we need to build the pricing table component and build the data summaries for the tables based on the backend summary objects, and the right hand panel (lots of changes to Bloom components here for the apply sections).

The review instructions I would recommend for this are to make sure the above list of data is showing as the real data from the existing listing detail page by comparing the test listing on the react version to the test listing on the angular version. I might also recommend reviewing this together so that I can walk through what is expected to be done so far.

### CODE REVIEW

File: app/javascript/modules/listings/DirectoryHelpers.tsx [hunk 1]
```diff
@@ -52,58 +48,6 @@ const headerClassNames = "text-base text-gray-700 border-b"
 
 const habitatForHumanity = "Habitat for Humanity"
 
-// Returns every status bar under the image card for one listing
```
Comment (emilyjablonski) [inline app/javascript/modules/listings/DirectoryHelpers.tsx:55]: I just moved these to a different file (SharedHelpers) - no changes

File: app/javascript/modules/listings/SharedHelpers.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import dayjs from "dayjs"
+import { areLotteryResultsShareable } from "../../util/listingStatusUtil"
+import { getReservedCommunityType } from "../../util/languageUtil"
+import RailsSaleListing from "../../api/types/rails/listings/RailsSaleListing"
+import RailsRentalListing from "../../api/types/rails/listings/RailsRentalListing"
+
+export type RailsListing = RailsSaleListing | RailsRentalListing
+
+// Returns every status bar under the image card for one listing
+export const getListingImageCardStatuses = (
```
Comment (emilyjablonski) [inline app/javascript/modules/listings/SharedHelpers.tsx:11]: These are from a different file (DirectoryHelpers) - no changes

**Comment by jacksongonzales** [comment]
when i try to navigate to [a listing](https://dahlia-web-full-pr-1605.herokuapp.com/listings/a0W0P00000F8YG4UAN?react=true) with `?react=true` I get the dreaded 'minified react error'

looks like something is expecting a string but receiving undefined?

**Comment by jacksongonzales** [comment]
> @jacksongonzales Can you test locally and let me know if you see the same?

@emilyjablonski navigating here on the branch still gives me an error: http://localhost:3000/listings/a0W4U00000KRHeEUAX?react=true

It looks like our version of bloom doesn't recognize `ContentAccordion`, which is throwing the undefined error

File: .eslintrc.js [hunk 3]
```diff
@@ -156,6 +156,7 @@ module.exports = {
     "unicorn/prevent-abbreviations": "off",
     "import/order": "off",
     "@typescript-eslint/no-empty-function": "off",
+    "unicorn/consistent-function-scoping": "off",
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply .eslintrc.js:159]: It was causing false positives

**Comment by emilyjablonski** [comment]
@jacksongonzales Locally did you `yarn install` ?

**Comment by jacksongonzales** [comment]
> @jacksongonzales Locally did you `yarn install` ? We might also benefit from pairing on this "offline" :)

agreed, let's investigate tmrw!

File: .eslintrc.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (christine-sfg) [reply .eslintrc.js:159]: ok, will look into turning this back on after refactor

**Comment by christine-sfg** [comment]
Tis working for me now locally and on the review app after the latest uptake. This is a really great first pass for hooking up the api! 

I think we can wrap up this story and continue polishing in subsequent stories.  Here are my testing notes.

? Listing card image: The image still has a pretty dark gradient on top.
x Listing card name
? Listing card reserved tags: Senior Building tag is left aligned instead of centered
? Address under listing card: Address is good. Link not functional yet
x Developer under listing card
? Orange reserved card above table: Reserved message displays, but text is different
x Listing preference cards with titles: Lottery Preference cards are displaying.
? Additional eligibility rules present if fields exist: Rule boxes are displaying. More link needs translation and some links not displaying as links
x Building selection criteria link
? All listing features: Missing unit features accordion
? Additional fees deposit range and costs not included string: Needs translations
? Additional information cards: content displaying as markup in Program Rules
? Right panel application deadline status: We can fix this via DAH-1113
? Right panel waitlist component with right waitlist numbers: Needs translations and styling
? Right panel information session events: Handle markup, format dates e.g. July 29th vs July 29
? Right panel open house events: Just date format
? Right panel open listing lottery events: Handle markup, not displaying for senior listings

**Review by christine-sfg — APPROVED** [review]

**Pull request merged**